pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryLocalBeforeReturn.xml

```diff
@@ -208,8 +208,8 @@ public class UnnecessaryLocalBeforeReturnFP {
     <test-code>
         <description>#310 UnnecessaryLocalBeforeReturn statement order does not matter</description>
         <rule-property name="statementOrderMatters">false</rule-property>
-        <expected-problems>1</expected-problems>
-        <expected-linenumbers>5</expected-linenumbers>
+        <expected-problems>2</expected-problems>
+        <expected-linenumbers>3,10</expected-linenumbers>
```

Comment:
I think, this is actually now correct - in both cases we can't be sure, that there are no side-effects. #310 added the property to make PMD ignore this and just assume, there are no side-effects, if there are no local vars handed in/out. But in the first example, even if compute() wouldn't there might be fields used by compute() and changed by markComputationDone(), which we are unaware of.

I'm not sure whether the ruleProperty "statementOrderMatters" still works as it was intended to and whether the property actually makes sense at all...

Reply:
> I'm not sure whether the ruleProperty "statementOrderMatters" still works as it was intended to and whether the property actually makes sense at all...

I don't think it does... I think we should assume that all methods do side effects, except maybe those named `get...`, otherwise we're prone to FPs

Reply:
I've created #3124